src/serverModel.ts

```diff
@@ -277,16 +277,23 @@ export default function ServerModel(
             let integrationAttributes: IntegrationAttributes = mpInstance._Store.integrationAttributes;
 
             // https://go.mparticle.com/work/SQDSDKS-5053
-            if (mpInstance._Helpers.getFeatureFlag && mpInstance._Helpers.getFeatureFlag(Constants.FeatureFlags.CaptureIntegrationSpecificIds)) {
-
-                // Attempt to recapture click IDs in case a third party integration
-                // has added or updated  new click IDs since the last event was sent.
-                mpInstance._IntegrationCapture.capture();
-                const transformedClickIDs = mpInstance._IntegrationCapture.getClickIdsAsCustomFlags();
-                customFlags = {...transformedClickIDs, ...customFlags};
-
-                const transformedIntegrationAttributes = mpInstance._IntegrationCapture.getClickIdsAsIntegrationAttributes();
-                integrationAttributes = {...transformedIntegrationAttributes, ...integrationAttributes};
+            if (mpInstance._Helpers.getFeatureFlag) {
+                const { FeatureFlags } = Constants;
+                const isCaptureIntegrationSpecificIdsV2 = (mpInstance._Helpers.getFeatureFlag(FeatureFlags.CaptureIntegrationSpecificIdsV2) as string) || '';
+                const normalizedValue = isCaptureIntegrationSpecificIdsV2.toLowerCase();
+                if (normalizedValue === 'all' || normalizedValue === 'roktonly') {
+                    mpInstance._IntegrationCapture.capture();
+                    const transformedClickIDs = mpInstance._IntegrationCapture.getClickIdsAsCustomFlags();
+                    customFlags = { ...transformedClickIDs, ...customFlags };
+                    const transformedIntegrationAttributes = mpInstance._IntegrationCapture.getClickIdsAsIntegrationAttributes();
+                    integrationAttributes = { ...transformedIntegrationAttributes, ...integrationAttributes };
+                } else if (mpInstance._Helpers.getFeatureFlag(FeatureFlags.CaptureIntegrationSpecificIds)) {
+                    mpInstance._IntegrationCapture.capture();
+                    const transformedClickIDs = mpInstance._IntegrationCapture.getClickIdsAsCustomFlags();
+                    customFlags = { ...transformedClickIDs, ...customFlags };
+                    const transformedIntegrationAttributes = mpInstance._IntegrationCapture.getClickIdsAsIntegrationAttributes();
+                    integrationAttributes = { ...transformedIntegrationAttributes, ...integrationAttributes };
```

Comment:
I think this can be a single if statement and simplified.  lines 285-289 are idential to 291-295, unless i'm mistaken. 

We don't need to check the normalized values, then check the other feature flag, and run the same logic.  You can check it all once with an or statement, and then run through lines 285-289 once.